Flepper.QueryBuilder/Utils/CacheUtils.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Linq;
+
+namespace Flepper.QueryBuilder.Utils
+{
+    internal class CacheUtils
+    {
+        private static readonly ConcurrentDictionary<Type, string[]> DtoProperties = new ConcurrentDictionary<Type, string[]>();
+
+        public static string[] GetDtoProperties<T>() where T : class
+        {
+            var type = typeof(T);
```

Comment:
In this method I think you can do something like the example below:

```
var type = typeof(T);

if (DtoProperties.TryGetValue(type, out string[] data)) return data;

return AddDtoPropertiesToCache(type);
```

The fact that we are inside the circle of confidence, the code below will never be executed, with that we can remove it:

```
if (!DtoProperties.TryGetValue(type, out string[] data))
{
     throw new Exception($"Can't find a cache entry for type {type.Name}");
}
```